lib/active_fedora/indexing/descendant_fetcher.rb

```diff
@@ -0,0 +1,104 @@
+module ActiveFedora
+  module Indexing
+    # Finds all descendent URIs of a given repo URI (usually the base URI).
+    #
+    # This is a slow and non-performant thing to do, we need to fetch every single
+    # object from the repo.
+    #
+    # The DescendantFetcher is also capable of partitioning the URIs into "priority" URIs
+    # that will be first in the returned list. These prioritized URIs belong to objects
+    # with certain hasModel models. This feature is used in some hydra apps that need to
+    # index 'permissions' objects before other objects to have the solr indexing work right.
+    # And so by default, the prioritized class names are the ones form Hydra::AccessControls,
+    # but you can alter the prioritized model name list, or set it to the empty array.
+    #
+    #     DescendentFetcher.new(ActiveFedora.fedora.base_uri).descendent_and_self_uris
+    #     #=> array including self uri and descendent uris with "prioritized" (by default)
+    #         Hydra::AccessControls permissions) objects FIRST.
+    #
+    # Change the default prioritized hasModel names:
+    #
+    #     ActiveFedora::Indexing::DescendantFetcher.default_priority_models = []
+    class DescendantFetcher
+      HAS_MODEL_PREDICATE = ::RDF::URI.new('info:fedora/fedora-system:def/model#hasModel').freeze
```

Comment:
Thanks, I suspected there was a constant somewhere, but didn't know how to find it.